.eslintrc.json

```diff
@@ -1,35 +1,43 @@
 {
-  "plugins": [
-    "react",
-    "react-hooks"
-  ],
-  "parserOptions": {
-    "ecmaVersion": 2018,
-    "sourceType": "module",
-    "ecmaFeatures": {
-      "jsx": true
+    "parser": "@typescript-eslint/parser",
+    "plugins": ["react", "react-hooks", "@typescript-eslint"],
+    "parserOptions": {
+        "ecmaVersion": 2018,
+        "sourceType": "module",
+        "ecmaFeatures": {
+            "jsx": true
+        }
+    },
+    "env": {
+        "browser": true,
+        "commonjs": true,
+        "es6": true,
+        "jest": true,
+        "node": true
+    },
+    "settings": {
+        "react": {
+            "version": "detect"
+        },
+        "import/resolver": {
+            "node": {
+                "extensions": [".js", ".ts", ".tsx"]
+            }
+        }
+    },
+    "extends": ["plugin:@typescript-eslint/recommended", "eslint:recommended", "plugin:react/recommended"],
+    "rules": {
+        "@typescript-eslint/explicit-function-return-type": "off",
+        "@typescript-eslint/no-explicit-any": "off",
+        "@typescript-eslint/indent": "off",
```

Comment:
Indentation is handled by prettier, but it would make sense to have only one place that's responsible for code formatting. I suggest adding prettier integration into eslint, and have eslint do all the checking: https://prettier.io/docs/en/integrating-with-linters.html#eslint

Reply:
as you always commit with prettier it's not really useful imho. And it will slow down eslint, this one being already slow.

Eslint is about rules, prettier about the style. We always run prettier when we commit. It would be redundant and if you have it inside your editor boring imho

Reply:
Hmm, maybe then it's specific to my workflow only. I like to have auto-format on save, not a big deal for me. The only problem is that some eslint rules overlap with prettier, so we gotta be careful about them, because once prettier formats, eslint might break (that's the reason I disabled the indent rule here). If that becomes a problem - https://github.com/prettier/eslint-config-prettier can ensure we don't have conflicts anyway...

Reply:
> I like to have auto-format on save

Yeah it's editor specific. 

+1 for eslint-config-prettier